### PR TITLE
Remove duplication and move templating to agent modules

### PR DESCRIPTION
**Pull request by Savidude**

### CODE REVIEW

**Review by ctienshi — APPROVED** [review]
LGTM

File: modules/apim_gateway/manifests/params.pp [hunk 1]
```diff
@@ -16,51 +16,37 @@
...
+  $template_list = [
+    'repository/conf/api-manager.xml',
+    'repository/conf/datasources/master-datasources.xml',
+    'repository/conf/carbon.xml',
+    'repository/conf/user-mgt.xml',
+    'repository/conf/axis2/axis2.xml',
+  ]
+
+  # Define file list
+  $file_list = []
```
Comment (ctienshi) [inline modules/apim_gateway/manifests/params.pp:34]: Why are we defining these empty file lists?

File: modules/apim_gateway/templates/carbon-home/bin/wso2server.sh.erb [hunk 2]
```diff
@@ -33,7 +33,7 @@
 # OS specific support.  $var _must_ be set to either true or false.
 #ulimit -n 100000
 
-export JAVA_HOME="<%= @java_home %>"
+export JAVA_HOME="<%= @java_symlink %>"
```
Comment (ctienshi) [inline modules/apim_gateway/templates/carbon-home/bin/wso2server.sh.erb:36]: Any special reason to name it as "java-symlink" ?

File: modules/apim_gateway/manifests/params.pp (hunk 1 above)
Comment (Savidude) [reply modules/apim_gateway/manifests/params.pp:34]: The user can define files they need to add to the setup. The files get added here: https://github.com/wso2/puppet-apim/pull/114/files#diff-f887e469de4b8a03342d0df119a077afR35

File: modules/apim_gateway/templates/carbon-home/bin/wso2server.sh.erb (hunk 2 above)
Comment (Savidude) [reply modules/apim_gateway/templates/carbon-home/bin/wso2server.sh.erb:36]: The parameter points to the path of a symlink pointing to JAVA_HOME

**Review by VimukthiPerera — APPROVED** [review]

**Pull request merged**